Classes/Utility/PathUtility.php

```diff
@@ -38,19 +38,26 @@ class PathUtility {
 	 * @return mixed
 	 */
 	public static function translatePath($path) {
-		if (is_array($path) == FALSE) {
-			$path = (0 === strpos($path, '/') ? $path : GeneralUtility::getFileAbsFileName($path));
+		if (is_string($path)) {
+			$path = GeneralUtility::isAbsPath($path) ? $path : GeneralUtility::getFileAbsFileName($path);
 			if (is_dir($path)) {
 				$path = realpath($path) . '/';
 			}
-			$path = GeneralUtility::fixWindowsFilePath($path);
-		} else {
-			foreach ($path as $key => $subPath) {
-				if (TRUE === in_array($key, self::$knownPathNames)) {
-					$path[$key] = self::translatePath($subPath);
-				}
-			}
+			return GeneralUtility::fixWindowsFilePath($path);
+		} elseif (is_array($path)) {
+			return array_map(array(__CLASS__, __METHOD__), $path);
 		}
-		return $path;
+		return NULL;
+	}
+
+	/**
+	 * Get fields from view configuration that are template paths
+	 *
+	 * @param array $viewConfiguration
+	 * @return array
+	 */
+	public static function getTemplatePathFields(array $viewConfiguration) {
```

Comment:
I see where you're going with this extra method, and I'm not opposed to adding new methods - but the `translatePaths` method is widely used also in dependencies and the behavior of it must be preserved. We've moved quite far from the original problem now and have accumulated some side effects. It's overkill to avoid checking if the path is in a known array.

We could consider adding a second argument to `translatePath` called `$translateAny` which defaults to `FALSE` and if set to `TRUE` skips the `in_array` check in the recursive part of `translatePath`. The default should be to work exactly as before.

Thanks for adding a unit test btw ;) I didn't even know we lacked that one.